### DISCUSSION TITLE
Stop propogation of request to next middleware if response returned from first middleware

### DISCUSSION THREAD

**Discussion by codertech120386** [fastapi/fastapi#8723] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request, Response
from starlette.responses import JSONResponse, RedirectResponse

app = FastAPI()

@app.middleware("http")
async def verify_user_agent(request: Request, call_next):
    if request.url.path == "/even-or-odd" and request.method == "GET":
        try:
            num = int(request.query_params.get("num"))
            if num % 2 == 0:
                return JSONResponse(content={
                    "isEven": True
                }, status_code=200)
            else:
                return JSONResponse(content={
                    "isEven": False
                }, status_code=200)
        except:
            return JSONResponse(content={
            "message": "we do not allow string numbers"
        }, status_code=400)
    else:
        response = await call_next(request)

        return response

    
    
@app.middleware("http")
async def api_route(request: Request, call_back):
    if request.url.path == "/api":
        return JSONResponse(content={"api": True}, status_code=200)
    


@app.get("/even-or-odd", response_class = RedirectResponse)
def home_route():
    return "http://localhost:8000/api"

@app.get("/api", response_class = Response)
def home_route():
    return {"api": True}
```

### Description

I am working on a problem where in I am creating a middleware that responds to one api endpoint "/even-or-odd: and if the provided api endpoint is different then we go to next middleware which will be the endpoint that listens to "/api" else we exit out from the first middleware.
But in this case even after receiving correct endpoint for Middleware 1 the request still enters Middleware 2 and I am not able to understand how to mitigate this

### Operating System

macOS

### Operating System Details

MacOs Big Sur

### FastAPI Version

0.70.0

### Python Version

3.9.9

**Comment by YuriiMotov** [comment, marked as answer]
The order in with you add middlewares matters. The last added middleware will be entered first.

So, you just need to change the order you declare your middlewares.

Read more: https://fastapi.tiangolo.com/tutorial/middleware/#multiple-middleware-execution-order
